apps/guardian-ui/src/components/setup/screens/setConfiguration/SetConfiguration.tsx

```diff
@@ -0,0 +1,314 @@
+
+export const SetConfiguration: React.FC<Props> = ({ next }: Props) => {
+  const { t } = useTranslation();
+  const {
+    state: {
+      role,
+      configGenParams,
+      myName: stateMyName,
+      password: statePassword,
+      numPeers: stateNumPeers,
+    },
+    api,
+    submitConfiguration,
+  } = useSetupContext();
+  const theme = useTheme();
+  const isHost = role === GuardianRole.Host;
+  const isSolo = role === GuardianRole.Solo;
+  const [myName, setMyName] = useState(stateMyName);
+  const [password, setPassword] = useState(statePassword);
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [passwordCheck, setPasswordCheck] = useState(Boolean);
+  const [hostServerUrl, setHostServerUrl] = useState('');
+  const [defaultParams, setDefaultParams] = useState<ConfigGenParams>();
+  const [federationName, setFederationName] = useState('');
+  const [metaFields, setMetaFields] = useState<[string, string][]>([['', '']]);
+  const [blockConfirmations, setBlockConfirmations] = useState('');
+  const [network, setNetwork] = useState('');
+  const [bitcoinRpc, setBitcoinRpc] = useState<BitcoinRpc>({
+    kind: '',
+    url: '',
+  });
+  const [mintAmounts, setMintAmounts] = useState<number[]>([]);
+  const [error, setError] = useState<string>();
+  const { onCopy, hasCopied } = useClipboard(password);
+  const [numPeers, setNumPeers] = useState(
+    stateNumPeers ? stateNumPeers.toString() : isSolo ? '1' : MIN_BFT_NUM_PEERS
+  );
+
+  const { isOpen, onOpen, onClose } = useDisclosure();
+
+  useEffect(() => {
+    const initStateFromParams = (params: ConfigGenParams) => {
+      setDefaultParams(params);
+      setFederationName(params.meta?.federation_name || '');
+
+      const meta = { federation_name: '', ...params.meta };
+      setMetaFields(Object.entries(meta));
+
+      const mintModule = getModuleParamsFromConfig(params, ModuleKind.Mint);
+      if (mintModule?.consensus?.mint_amounts) {
+        setMintAmounts(mintModule?.consensus.mint_amounts);
+      }
+
+      const walletModule = getModuleParamsFromConfig(params, ModuleKind.Wallet);
+      const walletConsensus = walletModule?.consensus;
+      if (walletConsensus?.finality_delay !== undefined) {
+        setBlockConfirmations(walletConsensus.finality_delay.toString());
+      }
+      if (walletConsensus?.network) {
+        setNetwork(walletConsensus.network);
+      }
+      if (walletModule?.local?.bitcoin_rpc) {
+        setBitcoinRpc(walletModule.local.bitcoin_rpc);
+      }
+    };
+
+    if (configGenParams === null) {
+      api
+        .getDefaultConfigGenParams()
+        .then(initStateFromParams)
+        .catch((err) => {
+          console.error(err);
+        });
+    } else {
+      initStateFromParams(configGenParams);
+    }
+  }, [configGenParams, api]);
+
+  // Update password when updated from state
+  useEffect(() => {
+    setPassword(statePassword);
+  }, [statePassword]);
+
+  const isValid: boolean = isHost
```

Comment:
nit: might be cleaner to define the sets of criteria as arrays, then run `arr.every()` to check all the conditions

Reply:
fixed